### PR TITLE
95 openpyxl selenium imports

### LINKED ISSUE #84: [Bug]: PyYAML Cython related regression bug
### Describe the Bug

Bug due to Cython 3 regression bug in PyYAML package. Suggesting to pin PyYAML to last known working version 5.3.1 or  [<5.4](https://github.com/yaml/pyyaml/issues/724#issuecomment-1638587228)

### Instructions to Reproduce the Bug

Run ```make install``` in any project depending on nesta_ds_utils.

### Traceback

```shell
Collecting pyyaml==6.0 (from nesta_ds_utils[s3]@ git+https://github.com/nestauk/nesta_ds_utils.git)
  Using cached PyYAML-6.0.tar.gz (124 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [48 lines of output]
      running egg_info
      writing lib/PyYAML.egg-info/PKG-INFO
      writing dependency_links to lib/PyYAML.egg-info/dependency_links.txt
      writing top-level names to lib/PyYAML.egg-info/top_level.txt
      Traceback (most recent call last):
        File "/opt/homebrew/Caskroom/miniconda/base/envs/ahl_out_of_home_analysis/lib/python3.8/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/opt/homebrew/Caskroom/miniconda/base/envs/ahl_out_of_home_analysis/lib/python3.8/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/opt/homebrew/Caskroom/miniconda/base/envs/ahl_out_of_home_analysis/lib/python3.8/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/private/var/folders/_b/gyjq8jzs7yj9w2jqg1b73wyc0000gn/T/pip-build-env-rxq4b0rx/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 355, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
        File "/private/var/folders/_b/gyjq8jzs7yj9w2jqg1b73wyc0000gn/T/pip-build-env-rxq4b0rx/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 325, in _get_build_requires
          self.run_setup()
        File "/private/var/folders/_b/gyjq8jzs7yj9w2jqg1b73wyc0000gn/T/pip-build-env-rxq4b0rx/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 341, in run_setup
          exec(code, locals())
        File "<string>", line 288, in <module>
        File "/private/var/folders/_b/gyjq8jzs7yj9w2jqg1b73wyc0000gn/T/pip-build-env-rxq4b0rx/overlay/lib/python3.8/site-packages/setuptools/__init__.py", line 107, in setup
          return distutils.core.setup(**attrs)
        File "/private/var/folders/_b/gyjq8jzs7yj9w2jqg1b73wyc0000gn/T/pip-build-env-rxq4b0rx/overlay/lib/python3.8/site-packages/setuptools/_distutils/core.py", line 185, in setup
          return run_commands(dist)
        File "/private/var/folders/_b/gyjq8jzs7yj9w2jqg1b73wyc0000gn/T/pip-build-env-rxq4b0rx/overlay/lib/python3.8/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
          dist.run_commands()
        File "/private/var/folders/_b/gyjq8jzs7yj9w2jqg1b73wyc0000gn/T/pip-build-env-rxq4b0rx/overlay/lib/python3.8/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
          self.run_command(cmd)
        File "/private/var/folders/_b/gyjq8jzs7yj9w2jqg1b73wyc0000gn/T/pip-build-env-rxq4b0rx/overlay/lib/python3.8/site-packages/setuptools/dist.py", line 1233, in run_command
          super().run_command(command)
        File "/private/var/folders/_b/gyjq8jzs7yj9w2jqg1b73wyc0000gn/T/pip-build-env-rxq4b0rx/overlay/lib/python3.8/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/private/var/folders/_b/gyjq8jzs7yj9w2jqg1b73wyc0000gn/T/pip-build-env-rxq4b0rx/overlay/lib/python3.8/site-packages/setuptools/command/egg_info.py", line 319, in run
          self.find_sources()
        File "/private/var/folders/_b/gyjq8jzs7yj9w2jqg1b73wyc0000gn/T/pip-build-env-rxq4b0rx/overlay/lib/python3.8/site-packages/setuptools/command/egg_info.py", line 327, in find_sources
          mm.run()
        File "/private/var/folders/_b/gyjq8jzs7yj9w2jqg1b73wyc0000gn/T/pip-build-env-rxq4b0rx/overlay/lib/python3.8/site-packages/setuptools/command/egg_info.py", line 549, in run
          self.add_defaults()
        File "/private/var/folders/_b/gyjq8jzs7yj9w2jqg1b73wyc0000gn/T/pip-build-env-rxq4b0rx/overlay/lib/python3.8/site-packages/setuptools/command/egg_info.py", line 587, in add_defaults
          sdist.add_defaults(self)
        File "/private/var/folders/_b/gyjq8jzs7yj9w2jqg1b73wyc0000gn/T/pip-build-env-rxq4b0rx/overlay/lib/python3.8/site-packages/setuptools/command/sdist.py", line 113, in add_defaults
          super().add_defaults()
        File "/private/var/folders/_b/gyjq8jzs7yj9w2jqg1b73wyc0000gn/T/pip-build-env-rxq4b0rx/overlay/lib/python3.8/site-packages/setuptools/_distutils/command/sdist.py", line 251, in add_defaults
          self._add_defaults_ext()
        File "/private/var/folders/_b/gyjq8jzs7yj9w2jqg1b73wyc0000gn/T/pip-build-env-rxq4b0rx/overlay/lib/python3.8/site-packages/setuptools/_distutils/command/sdist.py", line 336, in _add_defaults_ext
          self.filelist.extend(build_ext.get_source_files())
        File "<string>", line 204, in get_source_files
        File "/private/var/folders/_b/gyjq8jzs7yj9w2jqg1b73wyc0000gn/T/pip-build-env-rxq4b0rx/overlay/lib/python3.8/site-packages/setuptools/_distutils/cmd.py", line 107, in __getattr__
          raise AttributeError(attr)
      AttributeError: cython_sources
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

### LINKED ISSUE #89: [Feature]: PyPI release
### Slack Post

- [X] The team agrees on the need for this feature

### Describe the Feature

There's a need to replicate common dependencies for the ONS SRS environment. They only accept PyPI / Conda package installation.

Here's what's needed:
- [ ] A release workflow.yml
- [ ] Checking setup.py/.cfg / pyproject.toml conforms with release process
- [ ] Releasing based on latest dev for a first 0.1 version, with updated dependencies
- [ ] Expand coverage of unit-testing workflow to Python >3.10 (3.11 and 3.12)
- [x] Update and relax fixed versions for libraries that have breaking updates/ deprecation changes (i.e. PyYAML, Altair, Selenium, webdriver_manager, Pandas)

### Examples of this Feature

[ojd_daps_skills](https://github.com/nestauk/ojd_daps_skills/blob/dev/.github/workflows/release.yml) is an example PyPI published package

### LINKED ISSUE #92: [Bug]: (viz) Selenium & Webdriver Manager interdependency updates, altair-saver incompatible with altair 5, WebDriver dangling resource
### Describe the Bugs

1. `altair_saver` is [not fully compatible](https://altair-viz.github.io/user_guide/saving_charts.html#altair-saver) with Altair 5.0 API, it is recommended to use [vl-convert-python](https://altair-viz.github.io/user_guide/saving_charts.html#vl-convert).
2. Breaking changes introduced to [Chrome WebDriver](https://developer.chrome.com/docs/chromium/new-headless), multiple updates to combination of Selenium+WebDriver Manager. Support for older Chrome WebDrivers are being deprecated and will soon creep into cache-refreshed/ new installs.
3. When `nesta_ds_utils.viz.altair.saving.py: save()` is called without passing in a WebDriver object, `_google_chrome_driver_setup()` is called to instantiate a Chrome WebDriver, which is left running in the background with dangling resources/ refs.

The above two bugs are intermingled with wider API revamps in the respective dependencies. It is worthwhile to introduce these at `nesta_ds_utils`, then adapt to bugs discovered downstream (and review downstream 'product' maintenance).

### Instructions to Reproduce the Bug

1. Install `[dev]` extra, then `pip install --upgrade altair`, optionally `pip install --upgrade altair-viewer altair-saver`, then run `pytest`.
2. Same as above.
3. Change Selenium/WebDriverManager/Altair dependency versions with `pip install --upgrade dependency==another.version`, re-run `pytest`.

### Traceback from pytest
```
tests/cleaning/test_datetime_ops.py ..........                                                                                                                      [ 15%]
tests/loading_saving/test_S3.py ...............................                                                                                                     [ 64%]
tests/loading_saving/test_file_ops.py .....                                                                                                                         [ 71%]
tests/networks/test_build.py .............                                                                                                                          [ 92%]
tests/viz/altair/test_format.py .                                                                                                                                   [ 93%]
tests/viz/altair/test_saving.py F.F.                                                                                                                                [100%]

================================================================================ FAILURES =================================================================================
__________________________________________________________________________ test_save_png_altair ___________________________________________________________________________

    def test_save_png_altair():
        """Test that that figures saved as png exist."""
        fig = alt.Chart(pd.DataFrame()).mark_bar()
        path = "tests/temp/"
>       saving.save(fig, "test_fig", path)

tests/viz/altair/test_saving.py:13: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/opt/homebrew/Caskroom/miniconda/base/envs/ds_utils/lib/python3.10/site-packages/nesta_ds_utils/viz/altair/saving.py:120: in save
    _save_png(fig, path, name, scale_factor, driver)
/opt/homebrew/Caskroom/miniconda/base/envs/ds_utils/lib/python3.10/site-packages/nesta_ds_utils/viz/altair/saving.py:43: in _save_png
    alt_saver.save(
/opt/homebrew/Caskroom/miniconda/base/envs/ds_utils/lib/python3.10/site-packages/altair_saver/_core.py:172: in save
    return saver.save(fp=fp, fmt=fmt)
/opt/homebrew/Caskroom/miniconda/base/envs/ds_utils/lib/python3.10/site-packages/altair_saver/savers/_saver.py:121: in save
    content = self._serialize(fmt, "save")
/opt/homebrew/Caskroom/miniconda/base/envs/ds_utils/lib/python3.10/site-packages/altair_saver/savers/_selenium.py:284: in _serialize
    out = self._extract(fmt)
/opt/homebrew/Caskroom/miniconda/base/envs/ds_utils/lib/python3.10/site-packages/altair_saver/savers/_selenium.py:237: in _extract
    "vega-lite.js": get_bundled_script(
/opt/homebrew/Caskroom/miniconda/base/envs/ds_utils/lib/python3.10/site-packages/altair_viewer/_scripts.py:40: in get_bundled_script
    version_str = find_version(version, listing[package])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

version = '5.16.3', candidates = ['4.0.2', '4.8.1', '4.17.0'], strict_micro = False

    def find_version(
        version: Optional[str], candidates: List[str], strict_micro: bool = False
    ) -> str:
        """Find a matching version string given a list of candidate versions.
    
        Parameters
        ----------
        version : str or None
            The version to match. If None, the newest non-development version will be used.
        candidates : list
            The list of candidate versions to consider.
        strict_micro : bool
            If True, then ensure that the micro version matches exactly.
            If False (default), then let the micro version float to the newest version.
    
        Returns
        -------
        version : str
            The matching versions from the list of candidates.
    
        Raises
        ------
        NoMatchingVersions : if no matching version is found.
    
        Examples
        --------
        >>> find_version('3', ['3.1.0', '4.0.0', '4.1.0', '4.2.0-alpha0'])
        '3.1.0'
        >>> find_version('4', ['3.1.0', '4.0.0', '4.1.0', '4.2.0-alpha0'])
        '4.1.0'
        >>> find_version('4.0', ['3.1.0', '4.0.0', '4.1.0', '4.2.0-alpha0'])
        '4.0.0'
        >>> find_version(None, ['3.1.0', '4.0.0', '4.1.0', '4.2.0-alpha0'])
        '4.1.0'
        >>> find_version("4.2.0-alpha0", ['3.1.0', '4.0.0', '4.1.0', '4.2.0-alpha0'])
        '4.2.0-alpha0'
        """
        if not candidates:
            raise NoMatchingVersions("No candidate versions provided.")
        cand = sorted([Version(c) for c in candidates])
        if version is None:
            cand = sorted(c for c in cand if c.dev is None)
            if not candidates:
                raise NoMatchingVersions(f"No non-dev candidates in {candidates}.")
            return str(cand[-1])
    
        v = Version(version)
        if not strict_micro and not v.dev:
            v.micro = None
        matches = [c for c in cand if v.matches(c)]
        if not matches:
>           raise NoMatchingVersions(
                f"No matches for version={version!r} among {candidates}.\n"
                "Often this can be fixed by updating altair_viewer:\n"
                "    pip install -U altair_viewer"
            )
E           altair_viewer._utils.NoMatchingVersions: No matches for version='5.16.3' among ['4.0.2', '4.8.1', '4.17.0'].
E           Often this can be fixed by updating altair_viewer:
E               pip install -U altair_viewer

/opt/homebrew/Caskroom/miniconda/base/envs/ds_utils/lib/python3.10/site-packages/altair_viewer/_utils.py:212: NoMatchingVersions
__________________________________________________________________________ test_save_svg_altair ___________________________________________________________________________

    def test_save_svg_altair():
        """Test that that figures saved as svg exist."""
        fig = alt.Chart(pd.DataFrame()).mark_bar()
        path = "tests/temp/"
>       saving.save(fig, "test_fig", path, save_png=False, save_svg=True)

tests/viz/altair/test_saving.py:31: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/opt/homebrew/Caskroom/miniconda/base/envs/ds_utils/lib/python3.10/site-packages/nesta_ds_utils/viz/altair/saving.py:115: in save
    driver = _google_chrome_driver_setup() if driver is None else driver
/opt/homebrew/Caskroom/miniconda/base/envs/ds_utils/lib/python3.10/site-packages/nesta_ds_utils/viz/altair/saving.py:25: in _google_chrome_driver_setup
    driver = webdriver.Chrome(
/opt/homebrew/Caskroom/miniconda/base/envs/ds_utils/lib/python3.10/site-packages/selenium/webdriver/chrome/webdriver.py:70: in __init__
    super(WebDriver, self).__init__(DesiredCapabilities.CHROME['browserName'], "goog",
/opt/homebrew/Caskroom/miniconda/base/envs/ds_utils/lib/python3.10/site-packages/selenium/webdriver/chromium/webdriver.py:89: in __init__
    self.service.start()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <selenium.webdriver.chrome.service.Service object at 0x16dfe3040>

    def start(self):
        """
        Starts the Service.
    
        :Exceptions:
         - WebDriverException : Raised either when it can't start the service
           or when it can't connect to the service
        """
        try:
            cmd = [self.path]
            cmd.extend(self.command_line_args())
            self.process = subprocess.Popen(cmd, env=self.env,
                                            close_fds=system() != 'Windows',
                                            stdout=self.log_file,
                                            stderr=self.log_file,
                                            stdin=PIPE,
                                            creationflags=self.creationflags)
        except TypeError:
            raise
        except OSError as err:
            if err.errno == errno.ENOENT:
                raise WebDriverException(
                    "'%s' executable needs to be in PATH. %s" % (
                        os.path.basename(self.path), self.start_error_message)
                )
            elif err.errno == errno.EACCES:
                raise WebDriverException(
                    "'%s' executable may have wrong permissions. %s" % (
                        os.path.basename(self.path), self.start_error_message)
                )
            else:
                raise
        except Exception as e:
            raise WebDriverException(
                "The executable %s needs to be available in the path. %s\n%s" %
                (os.path.basename(self.path), self.start_error_message, str(e)))
        count = 0
        while True:
            self.assert_process_still_running()
            if self.is_connectable():
                break
    
            count += 1
            sleep(0.5)
            if count == 60:
>               raise WebDriverException("Can not connect to the Service %s" % self.path)
E               selenium.common.exceptions.WebDriverException: Message: Can not connect to the Service /Users/solomonyu/.wdm/drivers/chromedriver/mac64/120.0.6099.129/chromedriver-mac-arm64/chromedriver

/opt/homebrew/Caskroom/miniconda/base/envs/ds_utils/lib/python3.10/site-packages/selenium/webdriver/common/service.py:105: WebDriverException
============================================================================ warnings summary =============================================================================
../../../../../opt/homebrew/Caskroom/miniconda/base/envs/ds_utils/lib/python3.10/site-packages/botocore/compat.py:31
  /opt/homebrew/Caskroom/miniconda/base/envs/ds_utils/lib/python3.10/site-packages/botocore/compat.py:31: DeprecationWarning: 'urllib3[secure]' extra is deprecated and will be removed in a future release of urllib3 2.x. Read more in this issue: https://github.com/urllib3/urllib3/issues/2680
    from urllib3 import exceptions

tests/loading_saving/test_S3.py::test_upload_obj_unsup_data
  /opt/homebrew/Caskroom/miniconda/base/envs/ds_utils/lib/python3.10/site-packages/nesta_ds_utils/loading_saving/S3.py:238: UserWarning: Data uploaded as pickle. Please consider other accessible file types among the supported ones.
    warnings.warn(

tests/loading_saving/test_S3.py::test_download_obj_dict_geojson
  /opt/homebrew/Caskroom/miniconda/base/envs/ds_utils/lib/python3.10/site-packages/nesta_ds_utils/loading_saving/S3.py:392: UserWarning: Please check geojson has a member with the name 'type', the value of the member must be one of the following:'Point', 'MultiPoint', 'LineString', 'MultiLineString', 'Polygon', 'MultiPolygon', 'GeometryCollection','Feature' and 'FeatureCollection'. Else downloaded dictionary will not be valid geojson.
    warnings.warn(

tests/networks/test_build.py::test_network_as_adjacency_matrix
  /opt/homebrew/Caskroom/miniconda/base/envs/ds_utils/lib/python3.10/site-packages/nesta_ds_utils/networks/build.py:87: FutureWarning: adjacency_matrix will return a scipy.sparse array instead of a matrix in Networkx 3.0.
    return nx.adjacency_matrix(network), nodes

tests/networks/test_build.py::test_network_as_adjacency_matrix
  /opt/homebrew/Caskroom/miniconda/base/envs/ds_utils/lib/python3.10/site-packages/networkx/linalg/graphmatrix.py:173: DeprecationWarning: 
  
  The scipy.sparse array containers will be used instead of matrices
  in Networkx 3.0. Use `to_scipy_sparse_array` instead.
    return nx.to_scipy_sparse_matrix(G, nodelist=nodelist, dtype=dtype, weight=weight)

tests/viz/altair/test_format.py::test_nesta_theme_activation
tests/viz/altair/test_saving.py::test_save_png_altair
tests/viz/altair/test_saving.py::test_save_html_altair
  /opt/homebrew/Caskroom/miniconda/base/envs/ds_utils/lib/python3.10/site-packages/nesta_ds_utils/viz/altair/formatting.py:25: UserWarning: Averta font could not be located. Using 'Helvetica' instead
    warnings.warn("Averta font could not be located. Using 'Helvetica' instead")

tests/viz/altair/test_saving.py::test_save_png_altair
tests/viz/altair/test_saving.py::test_save_svg_altair
  /opt/homebrew/Caskroom/miniconda/base/envs/ds_utils/lib/python3.10/site-packages/nesta_ds_utils/viz/altair/saving.py:25: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
    driver = webdriver.Chrome(

tests/viz/altair/test_saving.py::test_save_png_altair
tests/viz/altair/test_saving.py::test_save_svg_altair
  /opt/homebrew/Caskroom/miniconda/base/envs/ds_utils/lib/python3.10/site-packages/nesta_ds_utils/viz/altair/saving.py:25: DeprecationWarning: use options instead of chrome_options
    driver = webdriver.Chrome(

tests/viz/altair/test_saving.py::test_save_png_altair
  /opt/homebrew/Caskroom/miniconda/base/envs/ds_utils/lib/python3.10/site-packages/selenium/webdriver/remote/remote_connection.py:390: DeprecationWarning: HTTPResponse.getheader() is deprecated and will be removed in urllib3 v2.1.0. Instead use HTTPResponse.headers.get(name, default).
    if response.getheader('Content-Type'):

tests/viz/altair/test_saving.py::test_save_png_altair
  /opt/homebrew/Caskroom/miniconda/base/envs/ds_utils/lib/python3.10/site-packages/selenium/webdriver/remote/remote_connection.py:391: DeprecationWarning: HTTPResponse.getheader() is deprecated and will be removed in urllib3 v2.1.0. Instead use HTTPResponse.headers.get(name, default).
    content_type = response.getheader('Content-Type').split(';')

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================================= short test summary info =========================================================================
FAILED tests/viz/altair/test_saving.py::test_save_png_altair - altair_viewer._utils.NoMatchingVersions: No matches for version='5.16.3' among ['4.0.2', '4.8.1', '4.17.0'].
FAILED tests/viz/altair/test_saving.py::test_save_svg_altair - selenium.common.exceptions.WebDriverException: Message: Can not connect to the Service /Users/solomonyu/....
=============================================================== 2 failed, 62 passed, 14 warnings in 36.89s ================================================================
```

### LINKED ISSUE #95: [Bug]: Import error for openpyxl and Selenium
### Describe the Bug

@j-gillam caught two import errors in review of #94 

1. Import error when reading/ writing .xlsx or .xlsm files
2. Import error with newest Selenium API in [viz] extras

### Instructions to Reproduce the Bug

- Try reading/ writing excel files
- Try saving Altair figures with the [viz] extras

### PR DESCRIPTION
**Pull request by sqr00t**
# Update some more dependencies

This PR closes the issues: #95 
This PR also partially addresses: #89 

### Description

1. Updates minimum version requirements for Selenium(>=4.17.2) and PyYAML(>=6.0.1)
2. Implement error handling for excel methods requiring `openpyxl` backend
3. Small changes to type hints for `path` parameters in function signatures in `viz.altair.saving.py`
4. Adds `__init__.py` to package modules

### Checklist:

- [x] I have followed [Contributor Guidelines](https://github.com/nestauk/nesta_ds_utils/blob/dev/CONTRIBUTING.md)
- [ ] I have updated the documentation to include any new functionality I have added or modified
- [ ] I have written unit tests for any new functionality I have added - not applicable
- [x] I have ran and passed all tests locally with `>> pytest`
- [x] I have checked that all tests ran successfully on the Github after I pushed

### CODE REVIEW

File: nesta_ds_utils/viz/altair/saving.py [hunk 1]
```diff
@@ -70,26 +74,32 @@ def _save_png(
     )
 
 
-def _save_html(fig: Chart, path: os.PathLike, name: str, scale_factor: int):
+def _save_html(
```
Comment (j-gillam) [inline nesta_ds_utils/viz/altair/saving.py:77]: Tiny tiny aesthetic comment so please feel free to ignore, but I noticed you separated out parameters into different lines for _save_svg and _save_png but haven't here.

**Review by j-gillam — APPROVED** [review]
Everything works for me! I trialled an excel example and saved a few altair plots.

File: nesta_ds_utils/viz/altair/saving.py (hunk 1 above)
Comment (sqr00t) [reply nesta_ds_utils/viz/altair/saving.py:77]: The pre-commit linting checks made these changes to separate lines if the line exceeds a (PEP8?) character count (should be ~80?)

File: nesta_ds_utils/loading_saving/__init__.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from nesta_ds_utils.loading_saving.gis_interface import _gis_enabled
```
Comment (j-gillam) [inline nesta_ds_utils/loading_saving/__init__.py:1]: Would it make sense to move the gis_interface try except into here as well? Instead of importing?

**Review by j-gillam — APPROVED** [review]
Worked perfectly in the example I have been testing it on. I have one comment around where the try excepts could be. But it isn't necessary for functionality, so happy for you to merge!

File: nesta_ds_utils/loading_saving/__init__.py (hunk 2 above)
Comment (sqr00t) [reply nesta_ds_utils/loading_saving/__init__.py:1]: I'd keep less logic in __init__.py , the imports are needed for defining the functions in gis_interface anyways. So I'd tend towards importing by traversing up the directory tree to the top level features_enabled dict

**Pull request merged**